client/blocks/reader-excerpt/index.jsx

```diff
@@ -115,16 +116,21 @@ const chooseExcerpt = ( post ) => {
 	return null;
 };
 
-const ReaderExcerpt = ( { post } ) => {
+const ReaderExcerpt = ( { post, setHasExcerpt } ) => {
 	const isDailyPrompt = !! getDailyPromptText( post );
+	const excerpt = chooseExcerpt( post );
+
+	useEffect( () => {
+		setHasExcerpt( !! excerpt );
```

Comment:
I think I may be getting an exception/crash when visiting `/read` due to `setHasExcerpt()` not being a function. If we haven't updated all callers to specify `setHasExcerpt`, then we should probably make this call conditional.
```suggestion
		setHasExcerpt?.( !! excerpt );
```

Reply:
Yep, that was it. Good catch, I apparently didn't check `/read`, only the `/discover` pages. 🙃